cli/command/task_test.go

```diff
@@ -0,0 +1,295 @@
+			EnvironmentID:      "env_id",
+			ContainerOverrides: overrides,
+		}
+
+		base.Resolver.EXPECT().
+			Resolve("deploy", "dpl_name").
+			Return(Args{"dpl_id"}, nil)
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name").
+			Return(Args{"env_id"}, nil)
+
+		base.Client.EXPECT().
+			CreateTask(req).
+			Return("job_id", nil)
+
+		if wait {
+			base.Client.EXPECT().
+				ReadJob("job_id").
+				Return(&models.Job{
+					Status: "Completed",
+					Result: "task_id",
+				}, nil)
+
+			base.Client.EXPECT().
+				ReadTask("task_id").
+				Return(&models.Task{}, nil)
+		}
+
+		c := NewContext(t, args, flags, SetNoWait(!wait))
+		if err := taskCommand.create(c); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestCreateTask_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing ENVIRONMENT arg": NewContext(t, nil, nil),
+		"Missing NAME arg":        NewContext(t, Args{"environment"}, nil),
+		"Missing DEPLOY arg":      NewContext(t, Args{"environment", "name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.create(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestDeleteTask(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Resolver.EXPECT().
+		Resolve("task", "task_name").
+		Return(Args{"task_id"}, nil)
+
+	base.Client.EXPECT().
+		DeleteTask("task_id").
+		Return("job_id", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "job_id",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("job_id").
+		Return(job, nil)
+
+	c := NewContext(t, Args{"task_name"}, nil)
+	if err := taskCommand.delete(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDeleteTask_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.delete(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestReadTask(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+	result := []*models.TaskSummary{
+		{TaskID: "task_id"},
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("task", "task_name").
+		Return(Args{"task_id"}, nil)
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return(result, nil)
+
+	base.Client.EXPECT().
+		ReadTask("task_id").
+		Return(&models.Task{}, nil)
+
+	c := NewContext(t, Args{"task_name"}, nil)
+	if err := taskCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTask_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Client.EXPECT().
```

Comment:
We need to change the implementation to do resolving / arg checking first. 
Part of the `userInputErrors` tests is checking that we don't 'waste' api calls when user input is wrong.